Client/Impl/Commands/UpdateJobTimeoutCommand.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using GatewayProtocol;
+using Zeebe.Client.Api.Commands;
+using Zeebe.Client.Api.Misc;
+using Zeebe.Client.Api.Responses;
+using UpdateJobTimeoutResponse = Zeebe.Client.Impl.Responses.UpdateJobTimeoutResponse;
+
+namespace Zeebe.Client.Impl.Commands;
+
+public class UpdateJobTimeoutCommand : IUpdateJobTimeoutCommandStep1, IUpdateJobTimeoutCommandStep2
+{
+    private readonly UpdateJobTimeoutRequest request;
+    private readonly Gateway.GatewayClient client;
+    private readonly IAsyncRetryStrategy asyncRetryStrategy;
+
+    public UpdateJobTimeoutCommand(Gateway.GatewayClient client, IAsyncRetryStrategy asyncRetryStrategy, long jobKey)
+    {
+        request = new UpdateJobTimeoutRequest()
+        {
+            JobKey = jobKey
+        };
+        this.client = client;
+        this.asyncRetryStrategy = asyncRetryStrategy;
+    }
+
+    public IUpdateJobTimeoutCommandStep2 Timeout(TimeSpan timeout)
+    {
+        request.Timeout = Convert.ToInt32(timeout.TotalMilliseconds);
```

Comment:
❓ Wondering about this: Why we converting to an int? Should be an long right? So Request.Timeout = (long)timeout.TotalMilliseconds; would be fine?

Reply:
https://github.com/camunda/zeebe/blob/main/zeebe/gateway-protocol/src/main/proto/gateway.proto#L555

Reply:
Yes, you're correct—it should indeed be a long. However, directly casting it might result in losing precision due to the double conversion of TotalMilliseconds. Given that we're dealing with milliseconds, the loss of precision may not significantly impact the outcome.

Reply:
Yes you are right. Casting it directly to long should be fine